### PR TITLE
Bug 1814192 - Set the delete menu item to inactive when there is no browsing data

### PR DESCRIPTION
**Pull request by gitstart**
### What
The PR will set the delete menu item to inactive when there is no browsing data

### Screenshots
| Issue video | Fix Video |
| ----------- | --------- |
|  https://user-images.githubusercontent.com/2725300/221115989-336e3206-794a-4988-8580-1c4a33b7ad2d.mp4 | https://www.loom.com/share/648f0e6f6abc4e95a0461b8ac318ca1d |

### Pull Request checklist
 - [X] Quality: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
 - [ ] Tests: This PR includes thorough tests or an explanation of why it does not
 - [ ] Changelog: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
 - [ ] Accessibility: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features
### After merge
- Milestone: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- Breaking Changes: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1814192

### CODE REVIEW

**Comment by t-p-white** [comment]
@gitstart thank you for your contribution! I would like to get some UX input on this, as the solution is ambiguous (disabled or removed)

@mozilla-mobile/ux do we want the button to be disabled or removed?

**Comment by lime124** [comment]
Hi! Thanks for your contribution! I agree it doesn't make sense to allow people to delete their history when there is no history to delete. 

Instead of hiding the button, the trash can should remain visible but instead be in the inactive state. Can you make that change please? Thanks!

**Comment by gitstart** [comment]
> Hi! Thanks for your contribution! I agree it doesn't make sense to allow people to delete their history when there is no history to delete.
> 
> Instead of hiding the button, the trash can should remain visible but instead be in the inactive state. Can you make that change please? Thanks!

@lime124 Should the inactive button have a different colour from the active button? If so which colour should it have? Thank you for getting back to me so quickly.

**Comment by Piratesarereal** [comment]
inactive/disabled state colour

For dark mode - rgba(251, 251, 254, 0.4);
For light mode - rgba(21, 20, 26, 0.4);

Users should not be able to interact with the icon button.

**Comment by amedyne** [comment]
@t-p-white when you have time to re-review this, this is good to go per review with UX and product.

File: fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 1]
```diff
@@ -207,6 +211,15 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
         }
     }
 
+    private fun updateDeleteMenuItemView(isEmpty: Boolean) {
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt:214]: nit: can we update 'isEmpty' to something like 'enabled' or ''isEnabled' then update callers as required

**Review by t-p-white — CHANGES_REQUESTED** [review]
Thanks again for your contribution! This is looking good 👍 once the comment is addressed, this is good to go

File: fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 2]
```diff
@@ -190,6 +193,7 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
 
         consumeFrom(historyStore) {
             historyView.update(it)
+            updateDeleteMenuItemView(it.isEmpty)
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt:196]:
```suggestion
            updateDeleteMenuItemView(!it.isEmpty)
```

File: fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 3]
```diff
@@ -207,6 +211,15 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
         }
     }
 
+    private fun updateDeleteMenuItemView(isEnabled: Boolean) {
+        if (isEnabled) {
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt:215]:
```suggestion
        if (!isEnabled) {
```

File: fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 4]
```diff
@@ -223,6 +236,8 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
                 }
         } else {
             inflater.inflate(R.menu.history_menu, menu)
+            deleteHistory = menu.findItem(R.id.history_delete)
+            updateDeleteMenuItemView(historyStore.state.isEmpty)
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt:240]:
```suggestion
            updateDeleteMenuItemView(!historyStore.state.isEmpty)
```

**Review by t-p-white — CHANGES_REQUESTED** [review]
Thank you for quickly addressing the previous comment. Apologies, I may not have been clear, but can you also update the callers to inverse the logic please? Essentially so the code reads as "disable the delete menu item if there is no history". The suggestions I've added will do that

**Comment by gitstart** [comment]
> 

Thank you @t-p-white for reviewing🙏.
Requested changes are pushed. 🚀

**Comment by t-p-white** [comment]
@gitstart no problem! This is great, thank you. Can you squash these 10 commits to a single commit please, and we can get this landed 🙏

**Comment by gitstart** [comment]
> @gitstart no problem! This is great, thank you. Can you squash these 10 commits to a single commit please, and we can get this landed 🙏

Hello @t-p-white,
Squashed commit pushed. Please take a look..

**Review by t-p-white — APPROVED** [review]
Thank you for your contribution @gitstart, nice work.

**Pull request merged**